### PR TITLE
FIX Mixed correlator IDs when cache events are used

### LINKED ISSUE #279: PEP: Race condition: Logs of diferent requests and transactions writing exactly the same correlatorId and transactionId
There some parts of the code were the "node domain" used to keep the correlatorId and the transactionId is not used in a proper way. That is causing that the logs of some requests and transactions used the id of another transactions. So in that cases there is no way to match the logs with the correct request and transaction.
This scenario it is got when the PEP receive two o more simultaneous requests.
We have detected that problem in this line of code:
https://github.com/telefonicaid/fiware-pep-steelskin/blob/master/lib/services/cacheUtils.js#L85
} else if (cache.updating[cacheType]) {
    cache.channel.on(cacheType, callback);
} else {
The request that is runing the method "cache.channel.on" is using a given "domain" with a given "corr" and "trans". But the "cacheType" event will be fired by the request that is resolving the request to the KeyStone. So the "callback" will write the logs using the wrong trans and corr id, the ones used by the other request.

https://jirapdi.tid.es/browse/DM-967

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #279

### CODE REVIEW

**Comment by fgalan** [comment]
LGTM (passing the ball to other reviewers for final LGTM + merge)

File: test/unit/request_correlator_test.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+                        done();
+                    });
+                });
+            });
+        });
+    }
+
+    var options = {
+        uri: 'http://localhost:' + config.resource.proxy.port + '/NGSI10/updateContext',
```
Comment (XavierVal) [inline test/unit/request_correlator_test.js]: Are this tests going to be launched always in localhost?

Comment (dmoranj) [inline test/unit/request_correlator_test.js]: Yes. The test launches a local instance of the PEP Proxy in order to execute the test. This instance is then instrumented changing its middlewares and decorating some functions (as you can see in the last test) that will be impossible (or extremely difficult) to test remotely.

**Comment by XavierVal** [comment]
LGTM

**Pull request merged**